python/ray/serve/backend_state.py

```diff
@@ -598,15 +600,14 @@ def __init__(self, controller_name: str, detached: bool,
         self._replica_constructor_retry_counter: Dict[BackendTag,
                                                       int] = defaultdict(int)
 
-        checkpoint = self._kv_store.get(CHECKPOINT_KEY)
-        if checkpoint is not None:
-            (self._replicas, self._backend_metadata,
-             self._backend_matadata_backup, self._deleted_backend_metadata,
-             self._target_replicas, self._target_versions,
-             self._backend_goals) = cloudpickle.loads(checkpoint)
+        self._resume_from_checkpoint()
+        if all_replica_tags and len(all_replica_tags) > 0:
```

Comment:
```suggestion
        if all_replica_tags is not None:
```

Passing in an empty list is valid.